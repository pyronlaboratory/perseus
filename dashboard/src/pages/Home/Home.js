import React from 'react'
import PropTypes from 'prop-types';

// material imports

import { withStyles } from '@material-ui/core/styles';

import AppBar from '@material-ui/core/AppBar';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

// dashboard components
import DataTable from '../../components/DataTable/DataTable'
import Insights from '../../containers/Insights/Insights';

const styles = (theme) => ({
  secondaryBar: {
    zIndex: 0,
  },
  wrapper: {
    padding: theme.spacing(6, 4),
    [theme.breakpoints.down('md')]: {
        padding: theme.spacing(3, 1),
    },
  }
});

// TO DO: Refactor: Move TabPanel in components directory
function TabPanel(props) {
    const { styling, children, value, index, ...other } = props;

    return (
      <div
        role="tabpanel"
        hidden={value !== index}
        id={`simple-tabpanel-${index}`}
        {...other}
      >
        {value === index && (
          <Box p={2}>
            {children}
          </Box>
        )}
      </div>
    );
}
  
TabPanel.propTypes = {
    children: PropTypes.node,
    index: PropTypes.any.isRequired,
    value: PropTypes.any.isRequired,
};
  
function Home(props) {
    const { classes } = props;
    const [value, setValue] = React.useState(0);

    const handleChange = (event, newValue) => {
        setValue(newValue);
    };
    return (
        <React.Fragment>
            <AppBar
                component="div"
                className={classes.secondaryBar}
                style={{ backgroundColor: "#18202c"}}
                position="static"
                elevation={0}
            >
                <Toolbar>
                <Grid container alignItems="center" spacing={1}>
                    <Grid item xs>
                    <Typography color="inherit" variant="h5" component="h1">
                        Dashboard
                    </Typography>
                    </Grid>
                </Grid>
                </Toolbar>
            </AppBar>

            <AppBar
                component="div"
                className={classes.secondaryBar}
                style={{ backgroundColor: "#18202c"}}
                position="static"
                elevation={0}
            >
                <Tabs value={value} onChange={handleChange} textColor="inherit">
                    <Tab textColor="inherit" label="Overview" /> 
                    <Tab textColor="inherit" label="Insights and Reporting" /> 
                </Tabs>
            </AppBar>
            
            <TabPanel className={classes.wrapper}  value={value} index={0}>
                <DataTable />
            </TabPanel>
            <TabPanel className={classes.wrapper} value={value} index={1}>
                <Insights />
            </TabPanel>

        </React.Fragment>
    )
}

Home.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Home);