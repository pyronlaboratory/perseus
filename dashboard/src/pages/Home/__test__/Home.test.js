import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Home from './../Home';

afterEach(cleanup);

describe('home page', () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Home />, div);
    })

    it("matches snapshot", () => {
        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
