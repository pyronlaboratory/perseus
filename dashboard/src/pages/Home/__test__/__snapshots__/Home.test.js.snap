// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page matches snapshot 1`] = `
Array [
  <div
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary Home-secondaryBar-1 MuiPaper-elevation0"
    style={
      Object {
        "backgroundColor": "#18202c",
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <h1
            className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
          >
            Dashboard
          </h1>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary Home-secondaryBar-1 MuiPaper-elevation0"
    style={
      Object {
        "backgroundColor": "#18202c",
      }
    }
  >
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Overview
            </span>
            <span
              className="PrivateTabIndicator-root-3 PrivateTabIndicator-colorSecondary-5 MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Insights and Reporting
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="Home-wrapper-2"
    hidden={false}
    id="simple-tabpanel-0"
    role="tabpanel"
  >
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <div
        className="MuiPaper-root makeStyles-paper-8 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          style={
            Object {
              "height": 400,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="Home-wrapper-2"
    hidden={true}
    id="simple-tabpanel-1"
    role="tabpanel"
  />,
]
`;
